### PR TITLE
FM2-133: Initial implementation of related person resource

### PR DESCRIPTION
**Pull request by corneliouzbett**
## Description of what I changed

Initial implementation of Related person resource

## Issue I worked on

see https://issues.openmrs.org/browse/FM2-133

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**.

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java [hunk 1]
```diff
@@ -75,6 +75,9 @@
 	
 	public static final String OPENMRS_FHIR_EXT_MEDICINE = OPENMRS_FHIR_EXT_PREFIX + "/medicine";
 	
+	public static final String OPENMRS_FHIR_EXT_RELATIONSHIP_IDENTIFIER = OPENMRS_FHIR_EXT_PREFIX
+	        + "/Relationship/identifier";
+
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java:80]: Why are we adding this new extension here?

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/RelatedPersonTranslatorImpl.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		
+		for (PersonAddress address : omrsRelatedPerson.getAddresses()) {
+			relatedPerson.addAddress(addressTranslator.toFhirResource(address));
+		}
+		
+		//identifier
+		Identifier relationshipIdentifier = new Identifier();
+		relationshipIdentifier.setSystem(FhirConstants.OPENMRS_FHIR_EXT_RELATIONSHIP_IDENTIFIER);
+		relationshipIdentifier.setValue(omrsRelatedPerson.getUuid());
+		relatedPerson.addIdentifier(relationshipIdentifier);
```
Comment (corneliouzbett) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/RelatedPersonTranslatorImpl.java:73]: > Why are we adding this new extension here?

I am adding an identifier to refer to person-A by uuid

File: api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java (hunk 1 above)
Comment (corneliouzbett) [reply api/src/main/java/org/openmrs/module/fhir2/FhirConstants.java:80]: I am adding an identifier to refer to person-A.

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/RelatedPersonTranslatorImpl.java (hunk 2 above)
Comment (ibacher) [reply api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/RelatedPersonTranslatorImpl.java:73]: I can't think of a better way than using an extension off-hand, but what I would do is 1. rename this to something like "RelatedPerson" and 2. change the data stored into a proper reference, e.g., `Person/<UUID>`, so it's a little more like a standard FHIR data type.

**Pull request merged**